app/plot_app/configured_plots.py

```diff
@@ -102,13 +102,47 @@ def generate_plots(ulog, px4_ulog, db_data, vehicle_data, link_to_3d_page,
                 vehicle_type = cur_dataset.data['vehicle_type']
                 in_transition_mode = cur_dataset.data['in_transition_mode']
                 vtol_states = []
-                for i in range(len(vehicle_type)):
-                    # a VTOL can change state also w/o in_transition_mode set
-                    # (e.g. in Manual mode)
-                    if i == 0 or in_transition_mode[i-1] != in_transition_mode[i] or \
```

Comment:
Given the i==0 check here it's not clear to me why `vtol_states` was previously `None`. Was there an exception ending up [here](https://github.com/PX4/flight_review/blob/4738c3777c1e113ca693ba076714f911f1b57ce3/app/plot_app/configured_plots.py#L159)?

Overall I'm confused, given your comment https://github.com/PX4/flight_review/issues/310#issuecomment-2899698120. Where is the problem exactly?
Do you have a log?

Did you see that the actual states are set in https://github.com/PX4/flight_review/blob/4738c3777c1e113ca693ba076714f911f1b57ce3/app/plot_app/configured_plots.py#L155 if not in transition?

I think using enums would make the code a bit clearer, but this is my fault in the first place.

Reply:
Here is a log: https://logs.px4.io/plot_app?log=7b635432-97f9-400a-91e7-8a2b67476875

Not sure if that helps, or whether you need the .ulg.

Reply:
> Can't we use [VtolStatus.msg](https://github.com/PX4/PX4-Autopilot/blob/main/msg/versioned/VtolVehicleStatus.msg) to distinguish between VTOL states in flight review?

Reply to myself after syncing with Beat: no, because VtolVehicleStatus.msg is not yet used by flightreview and we don't want to use new topics.